### PR TITLE
Request Code Review for Feature Branch

### PR DESCRIPTION
**Pull request by sddoe**
In this project we had created and implemented an exclusive clubhouse where your members can write anonymous posts. Inside the clubhouse, members can see who the author of a post is but, outside, they can only see the story and wonder who wrote it.

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work setting up this project. I was able to signup, create a post, and log out. Well done!
Let's make some changes to the PR so we can merge.
- [x] Add instructions for installing `gems`, creating and migrating database. One also needs to first install `yarn` before running `rails s` to avoid an error when running the app.
- [x] Change the title of your app from `Ruby: Enumerables` to something meaningful
- [x] The post route should allow specific actions: `resources :posts, only: [:new, :create, :index]` 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

### Martin Afani
#### Slack: whiz25

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -12,6 +12,23 @@
...
+    <% end %>
+
+    <% if member_signed_in? %>
+    <%= link_to current_member.email, edit_member_registration_path %>
+    <%= link_to 'logout', destroy_member_session_path, method: :delete %>
+
+    <% else %>
+    <%= link_to 'signin', new_member_session_path %>
+    <%= link_to 'sign up', new_member_registration_path %>
+    <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb]: There should be absolutely NO logic in views.

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      <tr>
+        <td><%= post.post %></td>
+        <% if member_signed_in? %>
+          <td><strong><%= post.member.name %></strong></td>
+          <td><%= link_to 'Show', post %></td>
+          <td><%= link_to 'Edit', edit_post_path(post) %></td>
+          <td><%= link_to 'Destroy', post, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        <% end %>
+      </tr>
+    <% end %>
```
Comment (acushlakoncept) [inline app/views/posts/index.html.erb]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @sddoe and @jamestucker1988, Great job! 👏  
You've done an awesome job so far 👌. But there are still some changes that are necessary for this project to be approved.

- [x] Your commit messages should be capitalized, and written in the imperative tense. You can watch this [video](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6) on how to edit your commit messages using git rebase

 See [Git Commit Best Practices](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) for more insights

- [x] After creating a post, It throws an error. 
Screen Shot 2021-03-15 at 5 41 48 PM

- [x] Your PR title should reflect what you did, Feature doesn't really say much

- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- User can sign in :+1: 
- Signed-in user can create new post :100: 
- Signed-in user can see post's author :muscle: 
- User can sign_out  :1st_place_medal: 
-  Guest can see only post's text :+1: 

To Improve :factory_worker:

- [ ] As I can see, your inline comments are helpful and make your code easy to read, but I suggest you focus on your code and logic. Using git history can help us to understand what you have done and  to review a previous version of your code, please, feel free to remove them.
- [x] It will be perfect if you remove also logic in your `application.html.erb` and put it in your helpers. As a developer your job is to create quality code that solves your client’s problems. Quality has many aspects to it, including readability and maintainability. It is much easier to read code that is lean and skinny, as well as serving a single purpose. When you look at a CSS file, you know it’s for styling and giving your site personality and uniqueness. When you look at a JavaScript file, you know it handles the interactivity on the front-end for the person viewing the site. And the view file is responsible for the HTML.
- [x] It will be nice if the Post form is not displayed when user is not signed-in in order to make your app more intuitive:
.

- [x] In your readme, It will be good if ask user to create DB via `rails db:create` before migrating.

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔✔**

This is a great project👏
Great work making the required changes.

You can merge and move on👌
Happy coding💻💻

**Pull request merged**